docs/release-process.md

```diff
@@ -1,177 +0,0 @@
-# Release Process
```

Comment:
Do we want to consider moving this to the Handbook together with other release processes https://handbook.mattermost.com/operations/research-and-development/product/release-process?

Reply:
I'd be okay with that, it was in here before and I was wondering if this belonged internally or not.

Reply:
I opened a PR here: https://github.com/mattermost/mattermost-handbook/pull/925.

Reply:
Great, I'll remove this file from the repo